test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/EnumerationTypeTests/Tests/EnumerationTypeQueryTests.cs

```diff
@@ -0,0 +1,649 @@
+        Assert.Equal(6, entity0["ProductID"]?.Value);
+        Assert.Equal("Mushrooms", entity0["Name"]?.Value);
+        Assert.Equal("Blue", (entity0["SkinColor"]?.Value as ODataEnumValue)?.Value);
+        Assert.Equal("ReadWrite", (entity0["UserAccess"]?.Value as ODataEnumValue)?.Value);
+
+        var entity1 = entries[1].Properties.ToDictionary(p => p.Name, p => p as ODataProperty);
+        Assert.Equal(7, entity1["ProductID"]?.Value);
+        Assert.Equal("Apple", entity1["Name"]?.Value);
+        Assert.Equal("Red", (entity1["SkinColor"]?.Value as ODataEnumValue)?.Value);
+        Assert.Equal("Read", (entity1["UserAccess"]?.Value as ODataEnumValue)?.Value);
+
+        var entity2 = entries[2].Properties.ToDictionary(p => p.Name, p => p as ODataProperty);
+        Assert.Equal(9, entity2["ProductID"]?.Value);
+        Assert.Equal("Computer", entity2["Name"]?.Value);
+        Assert.Equal("Green", (entity2["SkinColor"]?.Value as ODataEnumValue)?.Value);
+        Assert.Equal("Read", (entity2["UserAccess"]?.Value as ODataEnumValue)?.Value);
+    }
+
+    #endregion
+
+    #region Query entities order by enum property
+
+    [Theory]
+    [InlineData(MimeTypeODataParameterFullMetadata)]
+    [InlineData(MimeTypeODataParameterMinimalMetadata)]
+    [InlineData(MimeTypeApplicationAtomXml)]
+    public async Task QueryEntitiesOrderByEnumProperty(string mimeType)
+    {
+        // Arrange
+        var queryText = "Products?$orderby=SkinColor";
+
+        // Act
+        List<ODataResource> entries = await TestsHelper.QueryResourceSetsAsync(queryText, mimeType);
+
+        // Assert
+        Assert.Equal(5, entries.Count);
+
+        Assert.Equal(5, (entries[0].Properties.Single(p => p.Name == "ProductID") as ODataProperty)?.Value);
+        Assert.Equal(7, (entries[1].Properties.Single(p => p.Name == "ProductID") as ODataProperty)?.Value);
+        Assert.Equal(8, (entries[2].Properties.Single(p => p.Name == "ProductID") as ODataProperty)?.Value);
+        Assert.Equal(9, (entries[3].Properties.Single(p => p.Name == "ProductID") as ODataProperty)?.Value);
+        Assert.Equal(6, (entries[4].Properties.Single(p => p.Name == "ProductID") as ODataProperty)?.Value);
+    }
+
+    #endregion
+
+
+    #region Query entities select enum property
+
+    [Theory]
+    [InlineData(MimeTypeODataParameterFullMetadata)]
+    [InlineData(MimeTypeODataParameterMinimalMetadata)]
+    [InlineData(MimeTypeApplicationAtomXml)]
+    public async Task QueryEntitiesSelectEnumProperty(string mimeType)
+    {
+        // Arrange
+        var queryText = "Products?$select=SkinColor,UserAccess";
+
+        // Act
+        List<ODataResource> entries = await TestsHelper.QueryResourceSetsAsync(queryText, mimeType);
+
+        // Assert
+        Assert.Equal(5, entries.Count);
+
+        Assert.DoesNotContain(entries[0].Properties, p => p.Name != "SkinColor" && p.Name != "UserAccess");

```

Comment:
Should we also assert that it contains SkinColor and UserAccess? (Demonstrating that it doesn't contain properties that are not SkinColor nor UserAccess is not sufficient that prove that it contains SkinColor and UserAccess), it could contain no properties).